src/huggingface_hub/cli/repo.py

```diff
@@ -43,6 +47,41 @@
 
 logger = logging.get_logger(__name__)
 
+_SORT_PATTERN = re.compile(r"^(?P<key>[a-z_]+)(?::(?P<order>asc|desc))?$")
+_REPO_LIST_SORT_KEYS = {"last_modified", "trending_score", "created_at", "likes", "downloads"}
+
+_repo_list_sort_dict = {}
+for key in sorted(_REPO_LIST_SORT_KEYS):
+    _repo_list_sort_dict[key] = key
+    _repo_list_sort_dict[f"{key}_asc"] = f"{key}:asc"
+    _repo_list_sort_dict[f"{key}_desc"] = f"{key}:desc"
+
+RepoListSort = enum.Enum("RepoListSort", _repo_list_sort_dict, type=str, module=__name__)  # type: ignore
+
+
+def _repo_info_to_dict(info: Union[ModelInfo, DatasetInfo, SpaceInfo]) -> dict[str, object]:
+    """Convert repo info dataclasses to json-serializable dicts."""
+    created_at_str: str | None = None
+    if info.created_at is not None:
+        dt = info.created_at
+        if dt.tzinfo is None:
+            dt = dt.replace(tzinfo=timezone.utc)
+        else:
+            dt = dt.astimezone(timezone.utc)
+        created_at_str = dt.isoformat().replace("+00:00", "Z")
+
+    return {
+        "id": info.id,
+        "downloads": getattr(info, "downloads", None),
+        "likes": getattr(info, "likes", None),
+        "trendingScore": getattr(info, "trending_score", None),
+        "createdAt": created_at_str,
+        "private": getattr(info, "private", False),
+        "pipeline_tag": getattr(info, "pipeline_tag", None),
+        "library_name": getattr(info, "library_name", None),
+    }
```

Comment:
let's keep things simple here as well: convert dates to ISO format strings and call `dataclasses.asdict()`
```suggestion
    return {
        k: v.isoformat() if isinstance(v, datetime) else v
        for k, v in dataclasses.asdict(info).items()
        if v is not None
    }
```